File: app/src/main/java/activitytest/example/com/roomdemo/main/adapter/RecyclerviewAdapter.java


package activitytest.example.com.roomdemo.main.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import activitytest.example.com.roomdemo.R;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
public class RecyclerviewAdapter extends RecyclerView.Adapter<RecyclerviewAdapter.ViewHolder> {


    private final Context context;
    private final List<String> data;
    private final TextViewClickListener textViewClickListener;
    public RecyclerviewAdapter(Context context,List<String> data,TextViewClickListener textViewClickListener){
        this.context = context;
        this.data = data;
        this.textViewClickListener = textViewClickListener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate( R.layout.menu_item_card,parent,false);
        return new ViewHolder ( view );
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, final int position) {
        holder.name.setText(data.get(position));
        holder.name.setOnClickListener ( new View.OnClickListener () {
            @Override
            public void onClick(View v) {
                textViewClickListener.clickListener ( v );
            }
        } );

    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder{

        private final TextView name;

        public ViewHolder(View itemView) {
            super(itemView);
            name = itemView.findViewById(R.id.item);

        }
    }


    public interface TextViewClickListener{
        void clickListener(View textView);
    }
}

File: app/src/main/java/activitytest/example/com/roomdemo/main/adapter/RecyclerviewAdapter.kt
package activitytest.example.com.roomdemo.main.adapter

import activitytest.example.com.roomdemo.R
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class RecyclerviewAdapter(private val context: Context, private val data: List<String>, private val textViewClickListener: TextViewClickListener) : RecyclerView.Adapter<RecyclerviewAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.menu_item_card, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.name.text = data[position]
        holder.name.setOnClickListener { v -> textViewClickListener.clickListener(v) }
    }

    override fun getItemCount(): Int {
        return data.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val name: TextView = itemView.findViewById(R.id.item)

    }

    interface TextViewClickListener {
        fun clickListener(textView: View)
    }
}